examples/threading/threading.c

```diff
@@ -63,25 +68,23 @@ bool start_thread_obtaining_mutex(pthread_t *thread, pthread_mutex_t *mutex,int
      */
  
 
-    //allocate memory for thread_data
+    // allocate memory for thread_data
     struct thread_data *data = (struct thread_data*)malloc(sizeof(struct thread_data));
-    printf("thread_data mem allocation done\n");
+    DEBUG_LOG("thread_data memory allocation complete");
+    
     //set-up mutex and wait arguments
     data->mutex = mutex;
     data->wait_to_start_mutex = wait_to_obtain_ms;
     data->wait_to_release_mutex = wait_to_release_ms;
-    printf("mutex and wait arg set-up done\n");
-    //pass thread_data to create thread using threadfunc()
+    DEBUG_LOG("thread_data updated");
+    
+    // pass thread_data to create thread using threadfunc()
     int rc = pthread_create(thread, NULL, threadfunc, data);     
 
-    if(rc !=0 )
+    if(rc !=0 ){
+        ERROR_LOG("pthread_create: Thread creation failed");
```

Comment:
@ajsanthosh14  who frees data in this code path?

Reply:
@Maitreyee2095 the function description states

"* The start_thread_obtaining_mutex function should only start the thread and should not block
* for the thread to complete.
* The start_thread_obtaining_mutex function should use dynamic memory allocation for thread_data
* structure passed into the thread.  The number of threads active should be limited only by the
* amount of available memory.
* The thread started should return a pointer to the thread_data structure when it exits, which can be used
* to free memory as well as to check thread_complete_success for a successful exit."

According to this, we have to use the same data returned from the thread to check the success and to free the data, but to achieve that we may need to block for the thread to complete. In the function description, it says not to block for the thread to complete.

So I don't know how to free without waiting for the thread to complete? Please let me know is there any other way.
Thank you.